src/lib.rs

```diff
@@ -747,6 +747,11 @@ pub struct IterMut<'a, K: 'a, V: 'a> {
     marker: marker::PhantomData<(&'a K, &'a mut V)>,
 }
 
+/// A consuming insertion-order iterator over a `LinkedHashMap`'s entries.
+pub struct IntoIter<K, V, S> {
```

Comment:
No because we rely on the map to own every `Box<LinkedHashMapEntry<K, V>>`. If we don't keep the map, we need to move the data out of it first, or rework how the whole thing is represented so that the map no longer owns the data.